pilot/cmd/pilot-agent/status/server_test.go

```diff
@@ -553,6 +554,148 @@ func TestHttpsAppProbe(t *testing.T) {
 	}
 }
 
+func TestProbeHeader(t *testing.T) {
+	headerChecker := func(t *testing.T, header http.Header) *httptest.Server {
+		return httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
+			r.Header.Del("User-Agent")
+			r.Header.Del("Accept-Encoding")
+			if !reflect.DeepEqual(r.Header, header) {
+				t.Errorf("unexpected hedaer, want = %v, got = %v", header, r.Header)
+				http.Error(rw, "", http.StatusBadRequest)
+				return
+			}
+			http.Error(rw, "", http.StatusOK)
+		}))
+	}
+
+	testCases := []struct {
+		name          string
+		originHeaders http.Header
+		proxyHeaders  []apimirror.HTTPHeader
+		want          http.Header
+	}{
+		{
+			name: "Only Origin",
+			originHeaders: http.Header{
+				testHeader: []string{testHeaderValue},
+			},
+			proxyHeaders: []apimirror.HTTPHeader{},
+			want: http.Header{
+				testHeader: []string{testHeaderValue},
+			},
+		},
+		{
+			name: "Only Origin, has multiple values",
```

Comment:
what value does this add that the previous test case one doesn't? Doesn't it test the same thing?

Reply:
Avoid recurrence of the problem solved by the PR #28537